code/modules/mob/living/basic/guardian/guardian_creator.dm

```diff
@@ -71,144 +77,134 @@ GLOBAL_LIST_INIT(guardian_radial_images, setup_guardian_radial())
 		if(possible_guardian in possible_guardians)
 			continue
 		radial_options -= possible_guardian
-	var/mob/living/simple_animal/hostile/guardian/guardian_path
+	var/mob/living/basic/guardian/guardian_path
 	if(random)
 		guardian_path = pick(possible_guardians)
 	else
 		guardian_path = show_radial_menu(user, src, radial_options, custom_check = CALLBACK(src, PROC_REF(check_menu), user), radius = 42, require_near = TRUE)
-		if(!guardian_path)
+		if(isnull(guardian_path))
 			return
 	used = TRUE
 	to_chat(user, use_message)
-	var/guardian_type_name = "a random"
-	if(!random)
-		guardian_type_name = "the " + lowertext(initial(guardian_path.creator_name))
-	var/list/mob/dead/observer/candidates = poll_ghost_candidates("Do you want to play as [guardian_type_name] [mob_name] of [user.real_name]?", ROLE_PAI, FALSE, 100, POLL_IGNORE_HOLOPARASITE)
+	var/guardian_type_name = random ? "a random" : "the " + lowertext(initial(guardian_path.creator_name))
+	var/list/mob/dead/observer/candidates = poll_ghost_candidates(
+		"Do you want to play as [guardian_type_name] [mob_name] of [user.real_name]?",
```

Comment:
```suggestion
		"Do you want to play as [guardian_type_name] [mob_name] for [user.real_name]?",
```

may as well fix it this irks me

Reply:
"of" makes more sense, `Do you want to play the gravitokinetic guardian spirit of John` vs `Do you want to play the gravitokinetic guardian spirit for John`

Reply:
I'll just change it to `John's gravitokinetic guardian spirit`